include/boost/dll/detail/posix/shared_library_impl.hpp

```diff
@@ -105,18 +115,23 @@ class shared_library_impl {
         if (!!(mode & load_mode::append_decorations)) {
             mode &= ~load_mode::append_decorations;
 
-            boost::filesystem::path actual_path = (
-                std::strncmp(sl.filename().string().c_str(), "lib", 3)
-                ? (sl.has_parent_path() ? sl.parent_path() / L"lib" : L"lib").native() + sl.filename().native()
-                : sl
-            );
-            actual_path += suffix();
+            boost::filesystem::path actual_path = append_decorations(sl);
 
             handle_ = dlopen(actual_path.c_str(), static_cast<native_mode_t>(mode));
             if (handle_) {
                 boost::dll::detail::reset_dlerror();
                 return;
             }
+            boost::system::error_code prog_loc_err;
+            boost::filesystem::path loc = boost::dll::detail::program_location_impl(prog_loc_err);
+            if (boost::filesystem::exists(actual_path) && !boost::filesystem::equivalent(sl, loc, prog_loc_err)) {
```

Comment:
Looks like the same check should be added to the Windows implementation.

Reply:
A similar check is now present in the latest commit